staking/tests/split_vesting_account.ts

```diff
@@ -106,40 +116,177 @@ describe("split vesting account", async () => {
       { skipPreflight: true }
     );
 
-    let stakeAccount = await samConnection.getMainAccount(sam.publicKey);
+    let stakeAccount = await samConnection.getMainAccount(
+      samConnection.userPublicKey()
+    );
     assert(
       VestingAccountState.UnvestedTokensFullyUnlocked ==
         stakeAccount.getVestingAccountState(await samConnection.getTime())
     );
     await assertBalanceMatches(
       samConnection,
-      sam.publicKey,
+      samConnection.userPublicKey(),
       {
         unvested: {
-          unlocked: PythBalance.fromString("100"),
+          unlocked: PythBalance.fromString(vestingInitialBalance),
         },
       },
       await samConnection.getTime()
     );
+
+    let lockedPythBalance = PythBalance.fromString(lockedBalance);
+    if (lockedPythBalance.gt(PythBalance.zero())) {
+      // locking 0 tokens is an error
+      await samConnection.lockTokens(stakeAccount, lockedPythBalance);
+    }
+
+    let aliceConnection = await connect(new Keypair());
+    return [samConnection, aliceConnection];
+  }
+
+  async function assertMainAccountBalance(
+    samConnection: StakeConnection,
+    expectedState: VestingAccountState,
+    expectedBalance: OptionalBalanceSummary
+  ) {
+    let sourceStakeAccount = await samConnection.getMainAccount(
+      samConnection.userPublicKey()
+    );
+
+    assert(
+      expectedState ==
+        sourceStakeAccount.getVestingAccountState(await samConnection.getTime())
+    );
+    await assertBalanceMatches(
+      samConnection,
+      samConnection.userPublicKey(),
+      expectedBalance,
+      await samConnection.getTime()
+    );
+  }
+
+  it("split/accept flow success", async () => {
+    let [samConnection, aliceConnection] = await setupSplit("100", "100", "0");
+
+    let stakeAccount = await samConnection.getMainAccount(
+      samConnection.userPublicKey()
+    );
+
+    await samConnection.requestSplit(
+      stakeAccount,
+      PythBalance.fromString("33"),
+      aliceConnection.userPublicKey()
+    );
+
+    await pdaConnection.acceptSplit(
+      stakeAccount,
+      PythBalance.fromString("33"),
+      aliceConnection.userPublicKey()
+    );
+
+    await assertMainAccountBalance(
+      samConnection,
+      VestingAccountState.UnvestedTokensFullyUnlocked,
+      {
+        unvested: {
+          unlocked: PythBalance.fromString("67"),
+        },
+      }
+    );
+    await assertMainAccountBalance(
+      aliceConnection,
+      VestingAccountState.UnvestedTokensFullyUnlocked,
+      {
+        unvested: {
+          unlocked: PythBalance.fromString("33"),
+        },
+      }
+    );
```

Comment:
Wonder if we should check more stuff about Alice account

Reply:
Maybe add a call to updateVoterWeight

Reply:
I think this is fine as a test. You already have the tests in `voter_weight_test` around how the different types of tokens are translated into voter weights, so not sure what it achieves to do effectively the same test here.